File: tic_tac_toe.py
import random

board = [
    [0,0,0],
    [0,0,0],
    [0,0,0]
]

def print_board():
    print("\n   0  1  2")
    row_no=0
    for row in board:
        print(row_no, row)
        row_no+=1
    print("\n")

def is_empty( row, col):
    if board[row][col]==0:
        return True
    else:
        return False

def turn( player_no):
    valid=False
    
    while valid == False:
        row_no=int (input("row no: "))
        col_no=int (input("column no :"))
        if 0 <= row_no and row_no < 3 and 0 <= col_no and col_no < 3:
            if is_empty( row_no, col_no):
                valid=True
                board[row_no][col_no]=player_no
            else:
                print("Space Occupied\nTry again")
        else:
            print("\nWrong Input \nValid Input values [0,1,2]")
        
def m_turn(machine_no):
    valid=False

    while valid == False:
        row_no=int ( random.choice([0,1,2]) )
        col_no=int ( random.choice([0,1,2]))
        if 0 <= row_no and row_no < 3 and 0 <= col_no and col_no < 3:
            if is_empty( row_no, col_no):
                valid=True
                board[row_no][col_no]=machine_no

def is_winner(player_no):
    win=False

    for row in range(0,3):
      col1=board[row][0]
      col2=board[row][1]
      col3=board[row][2]
    if col1 == col2 == col3 == player_no:
        win = True
        return win

    for row in range(0,3):
      row1=board[0][row]
      row2=board[1][row]
      row3=board[2][row]
    if row1 == row2 == row3 == player_no:
        win = True
        return win

    d1=board[0][0]
    d2=board[1][1]
    d3=board[2][2]
    if d1 == d2 == d3 == player_no:
        win = True
        return win

    d1=board[0][2]
    d2=board[1][1]
    d3=board[2][0]
    if d1 == d2 == d3 == player_no:
        win = True
        return win


print("Welcome to Tic Tac Toe")
repeat = 'y'
while repeat == 'y' or repeat == 'Y':
    board = [
        [0,0,0],
        [0,0,0],
        [0,0,0]
    ]
    print_board()
    x=8
    m_choice=int (-1)
    valid = False
    while valid == False:
        choice =int (input("Choose what you want 1 or 2\n"))
        if choice == 1 or choice == 2:
            valid = True
        else:
            print("Invalid Input")

    if choice == 1:
        m_choice = 2    
    else:
        m_choice = 1

    print("your turn first")

    while(x):
        turn(choice)
        print_board()
        if x < 5:
            if is_winner(choice) or is_winner(m_choice):
                break
        x=x-1
        m_turn(m_choice)
        print_board()
        if x < 5:
            if is_winner(choice) or is_winner(m_choice):
                break
    
        x=x-1

    result=is_winner(choice)
    if result == True:
        print("You win!!!!!!")
    result = is_winner(m_choice)
    if result == True:
        print("You lost")
    if result == False:
        print("It was a draw")

    repeat = input("Want to play again? (y/n)")

File: Tic_Tac_Toe.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

bool is_empty(int a)
{
    if(a==0)
        return true;
    else
        return false;
}

void print_board(int board[3][3])
{
    printf("\n");
    for(int i=0;i<3;i++)
    {
        for(int j=0;j<3;j++)
        {
            printf("%d ", board[i][j]) ;
        }
        printf("\n");
    }
}

void move(int b[3][3],int r,int c,int ch)
{
    int i=0;
    while(i==0)
    {
        if( is_empty(b[r][c]) )
        {
            i=1;
            b[r][c]=ch;
        }
        else
        {
            printf("Place is already occupied. Try again");

        }
    }
}

int check_row(int ch,int board[3][3])
{
    int i,j;
    for(i=0;i<3;i++)
    {
        for(j=0;j<3;j++)
        {
            if(ch!=board[i][j])
            {
                break;
            }
        }
        if(j==3)
        {
            printf("You win");
            return 1;
    }
    else
        return 0;
    }
}

int check_col(int ch,int board[3][3])
{
    int i,j;
    for(i=0;i<3;i++)
    {
        for(j=0;j<3;j++)
        {
            if(ch!=board[j][i])
            {
                break;
            }
        }
        if(j==3)
        {
            printf("You win");
            return 1;
    }
    else
        return 0;
    }
}

int check_diagonal(int ch,int board[3][3])
{
    int i;
    for(i=0;i<3;i++)
    {
        if(board[i][i]!=ch)break;
    }
    for(i=0;i<3;i++)
    {
        if(board[i][2-i]!=ch)break;
    }
    if(i==3)
    {
        printf("You win");
        return 1;
    }
    else
        return 0;
}

int winner(int ch,int board[3][3])
{
    int result=0;
    if( (check_col(ch,board[3][3])==1) || (check_row(ch,board[3][3])==1) || (check_diagonal(ch,board[3][3])==1) )
        result=1;
    return result;
}

int main()
{
    int board[3][3]={0};
    print_board(board);
    int choice;
    printf("Enter your choice 1 or 2 : ");
    scanf("%d",&choice);
    int x=9;
    while(x--)
    {
        printf("Enter the number(for location) as on your numpad : ");
        int i;
        scanf("%d",&i);
        switch(i)
        {
            case 1:
        }
    }
    if(x==0)
    {
        pritnf("\nIt was a tie\nDo you want to play again?(y/n) :");
        char ch;
        scanf("%c",&ch);

    }
    return 0;
}
